spec/migrations/20190201173316_add_missing_ems_id_to_switch_spec.rb

```diff
@@ -1,35 +1,5 @@
 require_migration
 
-class AddMissingEmsIdToSwitch < ActiveRecord::Migration[5.0]
-  class ExtManagementSystem < ActiveRecord::Base
-    self.inheritance_column = :_type_disabled
-  end
-
-  class Host < ActiveRecord::Base
-    self.inheritance_column = :_type_disabled
-
-    belongs_to :ext_management_system, :foreign_key => "ems_id", :class_name => "AddMissingEmsIdToSwitch::ExtManagementSystem"
-
-    has_many :host_switches, :class_name => "AddMissingEmsIdToSwitch::HostSwitch"
-    has_many :switches, :through => :host_switches, :class_name => "AddMissingEmsIdToSwitch::Switch"
-  end
-
-  class HostSwitch < ActiveRecord::Base
-    belongs_to :host, :class_name => "AddMissingEmsIdToSwitch::Host"
-    belongs_to :switch, :class_name => "AddMissingEmsIdToSwitch::Switch"
-  end
-
-  class Switch < ActiveRecord::Base
-    self.inheritance_column = :_type_disabled
-
-    belongs_to :ext_management_system, :foreign_key => "ems_id", :class_name => "AddMissingEmsIdToSwitch::ExtManagementSystem"
-    belongs_to :host, :class_name => "AddMissingEmsIdToSwitch::Host"
-
-    has_many :hosts, :through => :host_switches, :class_name => "AddMissingEmsIdToSwitch::Host"
-    has_many :host_switches, :class_name => "AddMissingEmsIdToSwitch::HostSwitch"
-  end
-end
-
```

Comment:
There is no reason to re-open the migration class to define stubs... this was actually causing issues in the tests in rails 7 because it couldn't find the `has_many :host_switches, :class_name => "AddMissingEmsIdToSwitch::HostSwitch"` class `AddMissingEmsIdToSwitch::HostSwitch`.  It makes sense to move this to the migration class, even if the migration isn't using the stubs.

Reply:
The reason these are in the test only is because the migration itself doesn't use those classes.  At "runtime" we don't want to have to create stubs we don't need, because it can only introduce potential problems.

That being said, the way it was doing it before by reopening the migration is also wrong.  In other migrations where we need a "new" migration stub, we have a helper method called `new_migration_stub`.  Granted, it only creates a basic stub class, but it's possible we could pass a block to it to allow further manipulation.

Reply:
```
$ git grep "new_migration_stub"
spec/migrations/20200221200855_add_ancestry_to_miq_ae_namespace_spec.rb:4:  let(:ns_stub) { new_migration_stub(:miq_ae_namespaces) }
spec/migrations/20210519204247_move_miq_set_memberships_spec.rb:7:  let(:set_stub)        { new_migration_stub(:miq_sets) }
spec/migrations/20210519204247_move_miq_set_memberships_spec.rb:8:  let(:item_stub)       { new_migration_stub(:scan_items) }
spec/migrations/20210519204247_move_miq_set_memberships_spec.rb:9:  let(:rp_stub)         { new_migration_stub(:resource_pools) }
spec/migrations/20210824231354_add_timestamps_to_binary_blobs_spec.rb:4:  let(:binary_blob_stub) { new_migration_stub(:binary_blobs) }
spec/support/migration_helper.rb:62:      def new_migration_stub(table_name)
```

Reply:
yeah, we can possibly add the block case.  I don't see the potential problem of defining a migration stub within the migration class if it's only used in the test.  It feels wrong to be in the test itself unless it's a basic `new_migration_stub`.

Reply:
To be fair, our reuse of migration stubs through multiple tests are problematic under rails 7 so we may have to revisit this anyway.

Reply:
I'll drop this change from this PR.  I don't have a solution that I like for now.

Reply:
Oh wow - ok yeah - we'll have to discuss - sounds like we need to "reimplement" some of the stub handling for Rails 7 - what a pain.